### ISSUE TITLE
Whitespace not stripped from link preview card titles

### ISSUE THREAD

**Issue by daprice** [mastodon/mastodon#29993]
### Steps to reproduce the problem

1. View the post https://mastodon.social/@gvwilson/112294375846907813 and look at the title of the link preview card.

### Expected behaviour

Title should begin with text, not excessive whitespace

### Actual behaviour

Title contains excessive whitespace before the text

### Detailed description

The site this post links to has an… interesting way of marking up their title tags:
<img width="667" alt="image" src="https://github.com/mastodon/mastodon/assets/3615519/cb4e0848-523e-4d60-adaf-dff12d79fbfe">

While that is ultimately that site’s fault, Mastodon should probably strip the whitespace when parsing it for link previews. It doesn’t, so the result in the web UI is:
<img width="315" alt="image" src="https://github.com/mastodon/mastodon/assets/3615519/0ac6b12b-2ea7-4449-811f-6b967d32a301">

The string returned for the `card` by the Mastodon API (https://mastodon.social/api/v1/statuses/112294375846907813) is even worse:
`"title": "\r\n\t\t\r\n\t\t\r\n\t\t\t\r\n\t\t\t\r\n\t\t\r\n\t\tLab Dynamics: Management and Leadership Skills for Scientists, Third Edition\r\n\t"`

For comparison, it seems web browsers *do* strip the whitespace when displaying page titles.
<img width="301" alt="image" src="https://github.com/mastodon/mastodon/assets/3615519/d7785977-a4d2-47e8-a626-6c25b11e533b">

### Mastodon instance

mastodon.social

### Mastodon version

v4.3.0-nightly.2024-04-18

**Comment by timothyjrogers** [comment]
Started looking into this but I haven't been able to reproduce in my dev environment, either with latest from `main` or `version/v4.3.0-alpha.1`. The original site in question (cshlpress.com) seems to have fixed their weird `title` tag format, but I rehosted a similarly weird page on my website to link to. Looks normal in latest stable Edge, Firefox, and Chrome. I did confirm that the stored status still contains the literal carriage returns and tab characters.

 @daprice would it be possible to get some more details from you like what browser you were using when viewing this in and if you see the same in other browsers?

**Comment by daprice** [comment]
@timothyjrogers I was using Safari, and I can confirm that I only see the extra space in the web interface in that browser and not others. However, the same string is used for the `title` attribute, and I *do* see the extra whitespace in the tooltip when hovering over it in at least Firefox and Safari on macOS 14.4.1:

Firefox: 
<img width="631" alt="Screenshot 2024-04-20 at 6 04 05 PM" src="https://github.com/mastodon/mastodon/assets/3615519/ae150378-1141-455b-b671-b76bb1de9b5b">

Safari:
<img width="598" alt="Screenshot 2024-04-20 at 6 07 26 PM" src="https://github.com/mastodon/mastodon/assets/3615519/1eb11ace-7816-48ea-b4ad-3f43c54a9ad0">

**Comment by timothyjrogers** [comment]
Thanks for the additional details, @daprice ! I don't have a Mac handy to get a recent version of Safari, but I can reproduce the tooltip whitespace in Firefox at least. I've submitted a small change (PR link below) that fixes that when rendering the card component.

It would be straight forward to trim the whitespace when storing the status record initially, but I figured that might warrant more discussion. I don't know how important it is that that record faithfully reproduce the content that was extracted from the link.

PR: https://github.com/mastodon/mastodon/pull/30017
